test/jdk/javax/rmi/ssl/SSLSocketParametersTest.java

```diff
@@ -116,154 +130,104 @@ public ServerSocket createServerSocket(int port) throws IOException {
+                } catch (NoSuchAlgorithmException exc) {
+                    throw new RuntimeException("Could not create SSLContext.", exc);
+                } catch (IllegalArgumentException exc) {
+                    // expecting the exception for unsupported ciphersuite,
+                    // anything else is an error
+                    if (!exc.getMessage().toLowerCase().contains("unsupported ciphersuite")) {
+                        throw exc;
+                    }
+                }
             }
-            break;
-        case 4: /* server side dummy_ciphersuite */
-            System.out.println(msg1);
-            try {
-                HelloImpl server = new HelloImpl(
-                          0,
-                          new ClientFactory(),
-                          new ServerFactory(SSLContext.getDefault(),
-                                            new String[] {"dummy_ciphersuite"},
-                                            null,
-                                            false));
-                Remote stub = server.runServer();
-                HelloClient client = new HelloClient();
-                client.runClient(stub);
-                System.out.println(msg3);
-                System.exit(1);
-            } catch (Exception e) {
-                System.out.println(msg2 + " Exception: " + e.toString());
-                System.exit(0);
-            }
-            break;
-        case 5: /* server side dummy_protocol */
-            System.out.println(msg1);
-            try {
-                HelloImpl server = new HelloImpl(
-                          0,
-                          new ClientFactory(),
-                          new ServerFactory(null,
-                                            new String[] {"dummy_protocol"},
-                                            false));
-                Remote stub = server.runServer();
-                HelloClient client = new HelloClient();
-                client.runClient(stub);
-                System.out.println(msg3);
-                System.exit(1);
-            } catch (Exception e) {
-                System.out.println(msg2 + " Exception: " + e.toString());
-                System.exit(0);
+
+            /* server side dummy_protocol */
+            case 5 -> {
+                try {
+                    new ServerFactory(null,
+                            new String[]{"dummy_protocol"}, false);
+                    throw new RuntimeException(
+                            "The expected exception for unsupported protocol was not thrown.");
+                } catch (IllegalArgumentException exc) {
+                    // expecting the exception for unsupported protocol,
+                    // anything else is an error
+                    if (!exc.getMessage().toLowerCase().contains("unsupported protocol")) {
+                        throw exc;
+                    }
+                }
```

Comment:
Having exception handling in the run() method and also in this switch statement is quite confusing. It obscures the fact that these cases (4 and 5) don't call run() at all. It would be good to find a structure that makes it clear that some test cases are for exceptions that occur at ServerFactory creation time and other exceptions occur when an actual socket connection is attempted.

Reply:
Perhaps some refactoring should be done in a separate PR. 
The stated goal was to remove the use of the shell script; but its easy to start to refactor more and more.

Reply:
I did a little refactoring here to clarify (I hope) the tests and remove some redundant code.